samples/nrf9160/modem_shell/src/ppp/ppp_ctrl.c

```diff
@@ -39,6 +41,32 @@ int ppp_modem_data_socket_fd = PPP_MODEM_DATA_RAW_SCKT_FD_NONE;
 /* Socket to send and recv data to/from Zephyr PPP link: */
 int ppp_data_socket_fd = PPP_MODEM_DATA_RAW_SCKT_FD_NONE;
 
+/* Work queue for auto starting/stopping ppp according to default PDN
+ * activation status:
+ */
+struct ppp_ctrl_worker_data {
+	struct k_work work;
+	bool default_pdn_active;
+} ppp_ctrl_worker_data;
+
+/******************************************************************************/
+
+static void ppp_ctrl_link_default_pdn_status_handler(struct k_work *work_item)
+{
+	struct ppp_ctrl_worker_data *data_ptr =
+		CONTAINER_OF(work_item, struct ppp_ctrl_worker_data, work);
+
+	if (data_ptr->default_pdn_active == true) {
+		shell_info(shell_global,
+			   "Default PDN is active, running automatically: \"ppp up\"");
+		ppp_ctrl_start(shell_global);
+	} else {
+		shell_info(shell_global,
+			   "Default is PDN not active, running automatically: \"ppp down\"");
```

Comment:
Should this be "Default PDN is not active"?
I don't understand PPP implementation enough but check whether "running automatically" is fine here as it says "ppp down"

Reply:
Does this mean that what it's doing is the equivalent of doing "ppp up" or "ppp down" from the shell? This is fine with me, but it could also be something like "...starting PPP" and "...stopping PPP".

Reply:
I will finetune these texts correspondingly.